### PR TITLE
Add more test inputs for ArrayUtils.toMap() method

### PR DESCRIPTION
**Pull request by apirom9**
Add more test inputs for ArrayUtils.toMap() method as follow:
- zero-length array
- all null values array
- duplicate keys in array

### CODE REVIEW

File: src/test/java/org/apache/commons/lang3/ArrayUtilsTest.java [hunk 1]
```diff
@@ -264,6 +266,31 @@ public int hashCode() {
             }
         }});
         assertEquals("bar", map.get("foo"));
+
+        // Return empty map when got input array with length = 0
+        assertEquals(Collections.emptyMap(), ArrayUtils.toMap(new Object[0]));
+
+        // Test all null values
+        map = ArrayUtils.toMap(new Object[][] { {null, null}, {null, null} });
+        assertEquals(new HashMap<Object, Object>() {
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/lang3/ArrayUtilsTest.java]: Your expected map creation is too obtuse IMO, use the what Java provides instead:
```
Collections.singletonMap(null, null);
```

File: src/test/java/org/apache/commons/lang3/ArrayUtilsTest.java [hunk 2]
```diff
@@ -264,6 +266,31 @@ public int hashCode() {
...
+            {
+                put(null, null);
+            }
+        }, map);
+
+        // Test duplicate keys
+        map = ArrayUtils.toMap(new Object[][] { {"key", "value2"}, {"key", "value1"} });
+        assertEquals(new HashMap<Object, Object>() {
+
+            private static final long serialVersionUID = 1L;
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/lang3/ArrayUtilsTest.java]: See my previous comment.

File: src/test/java/org/apache/commons/lang3/ArrayUtilsTest.java (hunk 1 above)
Comment (apirom9) [reply src/test/java/org/apache/commons/lang3/ArrayUtilsTest.java]: Fixed. Thanks

File: src/test/java/org/apache/commons/lang3/ArrayUtilsTest.java (hunk 2 above)
Comment (apirom9) [reply src/test/java/org/apache/commons/lang3/ArrayUtilsTest.java]: Fixed. Thanks

**Pull request merged**